lib/src/working_copy.rs

```diff
@@ -138,6 +140,28 @@ pub struct TreeState {
     watchman_clock: Option<crate::protos::working_copy::WatchmanClock>,
 }
 
+impl Debug for TreeState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Self {
+            store: _,
+            working_copy_path,
+            state_path,
+            tree_id,
+            file_states: _,
+            sparse_patterns: _,
+            own_mtime,
+            watchman_clock,
+        } = self;
+        f.debug_struct("TreeState")
+            .field("working_copy_path", &working_copy_path)
+            .field("state_path", &state_path)
+            .field("tree_id", &tree_id)
+            .field("own_mtime", &own_mtime)
+            .field("watchman_clock", &watchman_clock)
+            .finish_non_exhaustive()
+    }
+}
```

Comment:
nit: can `#[derive(Debug)]`

Maybe it's too verbose to include in tracing, but I don't think it will always make sense to omit `file_states`, etc.

Reply:
@yuja the main problem is that it's too easy to accidentally slow everything down 100x because `file_states` will be large, particularly since `#[instrument]` doesn't make it obvious that `Debug` is being invoked. So you could simply add a parameter to a function that's annotated with `#[instrument]` and cause a bizarre performance regression that way. (And maybe the regression will only show up if your log level is set in a certain way?) I had to debug a performance regression like that once and it was really confusing.

Reply:
Understood. Thanks for the clarification.

I'm not sure where we should draw a line, but maybe `sparse_patterns` can be included?

Reply:
@yuja in my opinion we should omit it for now but the first person who needs it can add it 🙂.